Scalar.Common/FileBasedDictionary.cs

```diff
@@ -108,7 +110,7 @@ public void RemoveAndFlush(TKey key)
 
         public Dictionary<TKey, TValue> GetAllKeysAndValues()
         {
-            return new Dictionary<TKey, TValue>(this.data);
+            return new Dictionary<TKey, TValue>(this.data, this.comparer);
```

Comment:
We discovered a bug in the matching of keys in the 'config' dictionary; it was case sensitive when it's expected to be insensitive. The comparer is not passed to this _copy_ from the root `ConcurrentDictionary`.